### PR TITLE
Homework.lambda.part2.exercise2 (burachenko)

### PR DESCRIPTION
**Pull request by AlexMem**

### CODE REVIEW

**Review by Ilya21 — APPROVED** [review]

File: src/test/java/lambda/part2/exercise/Exercise2.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+@SuppressWarnings({"ConstantConditions", "unused"})
+class Exercise2 {
+
+    @Test
+    void personHasNotEmptyLastNameAndFirstName() {
+        // TODO предикат Person -> boolean, проверяющий что имя и фамилия человека не пусты
+        Predicate<Person> validate = person -> !person.getLastName().isEmpty() && !person.getFirstName().isEmpty();
```
Comment (vkovrovEPAM) [inline src/test/java/lambda/part2/exercise/Exercise2.java]: u can get null pointer exception, fix it

File: src/test/java/lambda/part2/exercise/Exercise2.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    // TODO - возвращает новый предикат, объединяющий исходные с помощью операции "AND"
+    // TODO - при реализации использовать логический оператор &&
+    private Predicate<Person> andUsingLogicalOperator(Predicate<Person> left, Predicate<Person> right) {
+        return person -> left.test(person) && right.test(person);
+    }
+
+    @Test
+    void personHasNotEmptyLastNameAndFirstNameUsingLogicalOperators() {
+        Predicate<Person> personHasEmptyFirstName = person -> person.getFirstName().isEmpty();
+        Predicate<Person> personHasEmptyLastName = person -> person.getLastName().isEmpty();
```
Comment (vkovrovEPAM) [inline src/test/java/lambda/part2/exercise/Exercise2.java]: same as above

File: src/test/java/lambda/part2/exercise/Exercise2.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    // TODO - возвращает новый предикат, объединяющий исходные с помощью операции "AND"
+    // TODO - при реализации использовать логический оператор &&
+    private <T> Predicate<T> and(Predicate<T> left, Predicate<T> right) {
+        return object -> left.test(object) && right.test(object);
+    }
+
+    @Test
+    void personHasNotEmptyLastNameAndFirstNameUsingGenericPredicates() {
+        Predicate<Person> personHasEmptyFirstName = person -> person.getFirstName().isEmpty();
+        Predicate<Person> personHasEmptyLastName = person -> person.getLastName().isEmpty();
```
Comment (vkovrovEPAM) [inline src/test/java/lambda/part2/exercise/Exercise2.java]: same here

File: src/test/java/lambda/part2/exercise/Exercise2.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+        assertThat(personHasNotEmptyLastNameAndFirstName.test(new Person("Алексей", "Доренко", 40)), is(true));
+        assertThat(personHasNotEmptyLastNameAndFirstName.test(new Person("Николай", "", 30)), is(false));
+        assertThat(personHasNotEmptyLastNameAndFirstName.test(new Person("", "Мельников", 20)), is(false));
+    }
+
+    @Test
+    void personHasNotEmptyLastNameAndFirstNameUsingStandardMethods() {
+        Predicate<Person> personHasEmptyFirstName = person -> person.getFirstName().isEmpty();
+        Predicate<Person> personHasEmptyLastName = person -> person.getLastName().isEmpty();
```
Comment (vkovrovEPAM) [inline src/test/java/lambda/part2/exercise/Exercise2.java]: same

**Review by vkovrovEPAM — CHANGES_REQUESTED** [review]

File: src/test/java/lambda/part2/exercise/Exercise2.java (hunk 1 above)
Comment (AlexMem) [reply src/test/java/lambda/part2/exercise/Exercise2.java]: fixed (4 times)

**Review by TarasKobzar — APPROVED** [review]

**Review by vkovrovEPAM — APPROVED** [review]

**Pull request merged**